src/cc2olx/tools/README.rst

```diff
@@ -0,0 +1,61 @@
+
+The ``directory`` argument is a directory containing video files that will be uploaded to edX's video encoding pipeline.
+The currently supported formats for the videos are ``mp4`` and ``mov``, which are the files supported by the edX video encoding pipeline.
+Note that files of other types will be ignored by the tool.
+
+The ``csv`` argument is a file that is meant to establish a relationship between a particular video, its externally hosted URL, and the edX video ID that is generated by the video encoding pipeline when the video is uploaded.
+
+In order to for this tool to run successfully, the following must be true of the CSV.
+
+* Each row in the file should describe a single video that appears in the directory supplied as a command line argument.
+* The CSV should contain a header row, labeling each column.
+* The first column should be labeled "Relative File Path". The values for this column should be the paths to the video files relative to the directory supplied as a command line argument. For example, if the directory is ``/Users/example/workspaces/videos`` and the video is located at ``/Users/example/workspaces/videos/unit_1/video_1.mp4``, the value of this cell should be ``unit_1/video_1.mp4``.
+* The CSV should contain the columns "External Video Link" and "Youtube ID". They can appear anywhere in the file as long as "Relative File Path" remains the first column.
+* Empty values should be represented as empty cells. Do not use "NONE", "N/A", etc.
+
+Output
+------
+
+The tool will generate a CSV. The name of the CSV will be the name of the original CSV supplied as a command line argument with "upload-results" appended to the end of the name, i.e. ``<original-filename>-upload-results.csv``.
+
+A new column, "Edx Id", which represents the edX Video ID for each video, will be appended to the end of the file.
+Otherwise, the file is identical to the file supplied as a command line argument.
+
+
+Logs
+----
+The tool will log to the console any errors that are encountered during execution.
+
+You can redirect the output to a file for later inspection.::
+
+    python src/cc2olx/tools/video_upload.py course-v1:edX+111222+111222 /Users/example/workspaces/videos/Users/example/workspaces/video-data.csv > logs.txt
+
+Use
+-----
+The video upload tool is a command-line Python 3 program.
+
+In order to make authenticated calls to Studio to generate signed URLs for uploading videos, you need to supply valid credentials.
+
+If you are an edX staff user, you can use the client credentials associated with a cc2olx oAuth2 application registered in the desired environment. If you are not an edX staff user, you will have to go through the process for generating a client ID and client secret. You can use the `instructions <https://course-catalog-api-guide.readthedocs.io/en/latest/authentication/#getting-a-client-id-and-client-secret
```

Comment:
To me, this is worth to be its own title and section.
It was not clear to me you expected the client_id and client_secret to be in the OS environment variables list.
Can we also have client_id and client_key optionally provided from the command line arguments?

Reply:
I can separate this section out into a separate section. What would be the benefit, though? Better visibility? I figure that the `Use` section is fairly short and the necessary information is there.

I decided against allowing credentials as command line arguments to avoid pasting a client secret in the user's terminal. My opinion is that it is risky to paste cleartext secrets in a terminal session. I figure that a little extra nuisance is worth the added little bit of security. What do you think?

Reply:
Ok, that is a consideration worth writing down here. I agree to leave my suggestion aside.
I believe the first user of this tool needs to figure out Auth info. This is the section they need to read and it would be important to highlight. Therefore my suggestion above. Not critical.

Reply:
That makes sense. I'll incorporate those changes